common/src/main/java/com/microsoft/identity/common/internal/controllers/BrokerMsalController.java

```diff
@@ -1170,11 +1173,29 @@ private void verifyTokenParametersAreSupported(@NonNull final TokenCommandParame
                     "The min broker protocol version for PopAuthenticationSchemeWithClientKey should be equal or more than 11.0."
                             + " Current required version is set to: " + parameters.getRequiredBrokerProtocolVersion());
         }
+
+        if (parameters.hasNestedAppParameters()) {
+            // Nested app auth is only supported when required protocol version is >= 15
+            if (!StringUtil.isNullOrEmpty(parameters.getChildClientId()) && !StringUtil.isNullOrEmpty(parameters.getChildRedirectUri())
```

Comment:
Should we validate these parameters at a different layer and here just do the broker protocol version check? For  most things, the parameter validation is happening inside the command parameters

Reply:
This is similar to other validations happening in this method. We want to verify here if the child params are sent, then the protocol has to be > 15.